### PR TITLE
Resetup all forking in place of behaviours

### PR DESCRIPTION
**Pull request by Alex-Swann**

### CODE REVIEW

File: apps/nrm/translations/src/en/fields.json [hunk 1]
```diff
@@ -333,6 +343,8 @@
...
     "div-1-paragraph-1" : "We will need to share their personal details with organisations such as the police and the Home Office if they agree to their case being referred.",
     "div-1-paragraph-2": "Please explain to the potential victim the ways in which the Home Office may process their personal information, including the purposes for which it will be used, the legal basis and who it will be shared with. This is contained in the <a class=\"govuk-link\" href=\"https://www.gov.uk/government/publications/human-trafficking-victims-referral-and-assessment-forms/privacy-information-notice-national-referral-mechanism\">Privacy Information Notice</a> and <a class=\"govuk-link\" href=\"https://www.gov.uk/government/organisations/home-office/about/personal-information-charter\">Personal Information Charter.</a>",
+    "automatic-referral": "As the potential victim either under 18 years old, or their age is unknown, this case will automatically be referred.",
+    "automatic-referral-div-1-paragraph-1" : "We will need to share their personal details with organisations such as the police and the Home Office if their case is referred.",
```
Comment (sulthan-ahmed) [inline apps/nrm/translations/src/en/fields.json:347]: this all makes sense, do we need to run this by Richard and co to explain the changes including the content?

File: apps/nrm/behaviours/save-form-session.js [hunk 2]
```diff
@@ -54,7 +54,9 @@ module.exports = superclass => class extends superclass {
           return res.redirect('/nrm/save-and-exit');
         }
 
-        if (req.path.match(/\/change$/)) {
+        const noEditContinue = !req.form.options.continueOnEdit;
+
+        if (req.sessionModel.get('redirect-to-reports') && noEditContinue) {
```
Comment (sulthan-ahmed) [inline apps/nrm/behaviours/save-form-session.js:59]: can you just explain what's going on here?

File: apps/nrm/behaviours/set-referral-state.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  saveValues(req, res, next) {
+    if (req.sessionModel.get('automatic-referral')) {
+      req.form.values['pv-want-to-submit-nrm'] = 'yes';
+    }
+
+    const isReferral = req.form.values['pv-want-to-submit-nrm'] === 'yes';
+
+    req.sessionModel.set('is-referral', isReferral);
```
Comment (sulthan-ahmed) [inline apps/nrm/behaviours/set-referral-state.js:18]: forgive me for ignorance, 
what is the need for `isReferral`? why can't we just use `automatic-referral`?

**Review by sulthan-ahmed — COMMENTED** [review]
Looks good, just a couple of questions.  Also, would it be worth demoing what you've done to Richard by releasing?
I think he'd be over the moon with the changes you've done 🙂

File: apps/nrm/translations/src/en/fields.json (hunk 1 above)
Comment (Alex-Swann) [reply apps/nrm/translations/src/en/fields.json]: yep all done

File: apps/nrm/behaviours/save-form-session.js (hunk 2 above)
Comment (Alex-Swann) [reply apps/nrm/behaviours/save-form-session.js:59]: a flag is set to redirect back to the reports in progress. This is to ensure it doesn't redirect to the latest question in progress or the check your answers page at the moment. It's intended to take the user back to whence they came.

File: apps/nrm/behaviours/set-referral-state.js (hunk 3 above)
Comment (Alex-Swann) [reply apps/nrm/behaviours/set-referral-state.js:18]: Yes this crossed my mind too. Automatic referral is only when the victim is under 18. isReferral is set for when an application needs to be sent as a referral and not DtN. The reason for having both is to ensure the integrity of the state of the application. 
```
Scenario 1 - Child
Automatic-referral : true
isReferral : true

Scenario 2 - Adult Referral
Automatic-referral : false
isReferral : true

Scenario 3 - Adult DtN
Automatic-referral : false
isReferral : false
```
As you can see they are not the same thing so both are required.

Comment (sulthan-ahmed) [reply apps/nrm/behaviours/set-referral-state.js:18]: Nice one @Alex-Swann

File: apps/nrm/translations/src/en/fields.json (hunk 1 above)
Comment (sulthan-ahmed) [reply apps/nrm/translations/src/en/fields.json]: and it was beautiful man 👍

File: apps/nrm/behaviours/save-form-session.js (hunk 2 above)
Comment (sulthan-ahmed) [reply apps/nrm/behaviours/save-form-session.js:59]: 👍

**Pull request merged**